### ISSUE TITLE
Improve Typescript types / Make GlueSchemaRegistry multi-schema

### ISSUE THREAD

**Issue by Nopik** [meinestadt/glue-schema-registry#124]
Currently the `<T>` template parameter is defined on whole registry, and it basically forbids using single registry instance for several schemas. It should be moved to encode/decode. I.e.:

```
class GlueSchemaRegistry<T> {
  encode() {}
}
```

becomes

```
class GlueSchemaRegistry {
  encode<T>() {}
}
```

so registry can be used for several types.

**Comment by felix-iw** [comment]
Hi Nopik, Thank you for bringing this up. 
I would like to understand your use case a bit better. Do you have a real use case where you work with different schemas where being able to work with the same instance of GlueSchemaRegistry would be beneficial?
Is there any reason why you don't want to use two dedicated instances instead?
Like:

```typescript
const schema1 = new GlueSchemaRegistry<ISchema>(...);
const schema2 = new GlueSchemaRegistry<ISchema2>(...);
```

**Comment by Nopik** [comment]
Definitely. I have scenario, where I do have 1 schema registry containing multiple schemas, 1 for each of the data types I do have in my processing pipeline.

Glue Schema Registry in general do support multiple schemas per registry, so I'm not sure why there would be any limitation to 1 schema per registry.

**Comment by felix-iw** [comment]
Yes, but is there any good reason why you don't want to create separate instances per schema?
Having one instance for multiple, independent schemas, might be confusing and more difficult to read in the code. But, to be consistent, it might be a good idea to specify the schema name in the constructor, and make sure that register only registers a new version for that schema.

**Comment by Nopik** [comment]
The way I have the code organized is that there is some central 'schema service' containing all schemas in my project, so whenever any other part in the code needs to decode something, it only passes schema name to the service's `decode` function, which uses registry internally. Making multiple registries in such scenario makes little sense.

Additionally, are you 100% sure that multiple registries are optimal from resource point of view? That AWS SDK doesn't reuse connections under the hood, or obtain some authorization tokens per instance? And never will? :) Like AWS Lambda client does utilize some internal caching a lot when you're invoking lambdas in large amounts, having multiple instances of that would be much less optimal.
